scripts/scil_bundle_label_map.py

```diff
@@ -86,224 +149,205 @@ def _build_arg_parser():
+        cut_sft.data_per_point['color'] = ArraySequence(cut_sft.streamlines)
         if not os.path.isdir(sub_out_dir):
             os.mkdir(sub_out_dir)
 
-        # Save each session map if multiple inputs
-        nib.save(nib.Nifti1Image((binary_list[i]*labels_map).astype(np.uint16),
-                                 sft_list[0].affine),
-                 os.path.join(sub_out_dir, 'labels_map.nii.gz'))
-        nib.save(nib.Nifti1Image(binary_list[i]*distance_map,
-                                 sft_list[0].affine),
-                 os.path.join(sub_out_dir, 'distance_map.nii.gz'))
-        nib.save(nib.Nifti1Image(binary_list[i]*corr_map,
-                                 sft_list[0].affine),
-                 os.path.join(sub_out_dir, 'correlation_map.nii.gz'))
-
-        if len(sft):
-            tmp_labels = ndi.map_coordinates(labels_map,
-                                             sft.streamlines._data.T-0.5,
-                                             order=0)
-            tmp_dists = ndi.map_coordinates(distance_map,
-                                            sft.streamlines._data.T-0.5,
-                                            order=0)
-            tmp_corr = ndi.map_coordinates(corr_map,
-                                           sft.streamlines._data.T-0.5,
-                                           order=0)
-            cmap = plt.colormaps[args.colormap]
-            new_sft.data_per_point['color'] = ArraySequence(
-                new_sft.streamlines)
-
-            # Nicer visualisation for MI-Brain
-            new_sft.data_per_point['color']._data = cmap(
-                tmp_labels / np.max(tmp_labels))[:, 0:3] * 255
-        save_tractogram(new_sft,
-                        os.path.join(sub_out_dir, 'labels.trk'))
-
-        if len(sft):
-            new_sft.data_per_point['color']._data = cmap(
-                tmp_dists / np.max(tmp_dists))[:, 0:3] * 255
-        save_tractogram(new_sft,
-                        os.path.join(sub_out_dir, 'distance.trk'))
-
-        if len(sft):
-            new_sft.data_per_point['color']._data = cmap(tmp_corr)[
-                :, 0:3] * 255
-        save_tractogram(new_sft,
-                        os.path.join(sub_out_dir, 'correlation.trk'))
+        # Dictionary to hold the data for each type
+        data_dict = {'labels': labels_map.astype(np.uint16),
+                     'distance': distance_map.astype(np.float32),
+                     'correlation': corr_map.astype(np.float32)}
+
+        # Iterate through each type to save the files
+        for basename, map in data_dict.items():
+            nib.save(nib.Nifti1Image((binary_list[i] * map), sft_list[0].affine),
+                     os.path.join(sub_out_dir, f'{basename}_map.nii.gz'))
+
+            if basename == 'correlation' and len(args.in_bundles) == 1:
+                continue
+
+            if len(sft):
+                tmp_data = ndi.map_coordinates(
```

Comment:
Same comment, careful with map_coordinates. Everything under the if needs more comments and could be moved into a function as well.